csp_collector.go

```diff
@@ -36,33 +37,96 @@ var (
 
 	// Flag for toggling verbose output.
 	debugFlag bool
+
+	// Path to file which has blocked URI's per line
+	blockedURIfile string
+
+	// Default URI Filter list
+	ignoredBlockedURIs = []string{
+		"resource://",
+		"chromenull://",
+		"chrome-extension://",
+		"safari-extension://",
+		"mxjscall://",
+		"webviewprogressproxy://",
+		"res://",
+		"mx://",
+		"safari-resource://",
+		"chromeinvoke://",
+		"chromeinvokeimmediate://",
+		"mbinit://",
+		"opera://",
+		"localhost",
+		"127.0.0.1",
+		"none://",
+		"about:blank",
+		"android-webview",
+		"ms-browser-extension",
+		"wvjbscheme://__wvjb_queue_message__",
+		"nativebaiduhd://adblock",
+		"bdvideo://error",
+	}
+
+	// TCP Port to listen on
+	listenPort int
 )
 
 func setupDebugLogger(debugHandle io.Writer) {
 	debug = log.New(debugHandle, "[DEBUG] ", log.Lmicroseconds)
 }
 
+func trimEmpty (s []string) []string {
+    var r []string
+    for _, str := range s {
+        if str != "" {
+            r = append(r, str)
+        }
+    }
+    return r
+}
+
 func main() {
 	setupDebugLogger(os.Stdout)
 
 	version := flag.Bool("version", false, "Display the version")
 	flag.BoolVar(&debugFlag, "debug", false, "Output additional logging for debugging")
+	flag.StringVar(&blockedURIfile, "filter-file", "", "Blocked URI Filter file")
+	flag.IntVar(&listenPort, "port", 8080, "Port to listen on")
 
 	flag.Parse()
 
 	if *version {
 		fmt.Printf("csp-collector (%s)\n", Rev)
-		os.Exit(0)
+	}
+
+	if len(blockedURIfile) > 0 {
+		content, err := ioutil.ReadFile(blockedURIfile)
+		if err != nil {
+			fmt.Printf("Error reading Blocked File list : %s", blockedURIfile)
+		}
+		ignoredBlockedURIs = trimEmpty(strings.Split(string(content), "\n"))
 	}
 
 	if debugFlag {
 		debug.Println("Starting up...")
+		debug.Printf("Using Filter list from : %s\n", blockedURIfile)
+		debug.Printf("Blocked URI List : %s", ignoredBlockedURIs)
+// Alternate Block list layout
+/*
+		debug.Println("--- START Blocked URIs ---")
+		for _, value := range ignoredBlockedURIs {
+			debug.Println("  ", value)
+		}
+		debug.Println("--- FINISH Blocked URIs ---")
+*/
+		debug.Printf("Listening on TCP Port : %v", listenPort)
 	}
 
 	http.HandleFunc("/", handleViolationReport)
-	log.Fatal(http.ListenAndServe(":8080", nil))
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%v", listenPort), nil))
```

Comment:
Similar comment to https://github.com/jacobbednarz/go-csp-collector/pull/10/files#r231663247.

```suggestion
	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", strconv.Itoa(listenPort)), nil))
```

Reply:
Done